### PR TITLE
Enh pydra run error msg

### PR DESCRIPTION
**Pull request by omar-rifai**
Small enhancement to the pydra run function to print error message. For debugging purpose.

### CODE REVIEW

File: clinica/pydra/engine_utils.py [hunk 1]
```diff
@@ -49,18 +49,27 @@ def bids_query(keys: list) -> dict:
 
 
 def run(wf: Workflow) -> None:
-
-    """Execute a Pydra workflow
```
Comment (NicolasGensollen) [inline clinica/pydra/engine_utils.py:53]: Why did you remove the docstring?

File: clinica/pydra/engine_utils.py [hunk 2]
```diff
@@ -49,18 +49,27 @@ def bids_query(keys: list) -> dict:
...
+    results = wf.result(return_inputs=False)
+    return results
+
+
+def read_error(path):
+    import cloudpickle as cp
+
+    with open(path, "rb") as fp:
+        err = cp.load(fp)
+        print(err["error message"])
```
Comment (NicolasGensollen) [inline clinica/pydra/engine_utils.py:75]: I would either return the error, or change the name of this function to `print_error`. WDYT?

**Review by NicolasGensollen — COMMENTED** [review]
Thanks @omar-rifai ! This will make debugging easier indeed ! 👍 
Couple suggestions below.

**Comment by omar-rifai** [comment]
Thanks @NicolasGensollen, I was thinking this could be temporary for debugging purposes but might as well have the right decorations. 

Should be good now.

File: clinica/pydra/engine_utils.py [hunk 3]
```diff
@@ -48,19 +52,53 @@ def bids_query(keys: list) -> dict:
...
-    with Submitter(plugin="cf") as submitter:
-        submitter(wf)
+    import cloudpickle as cp
 
-    results = wf.result(return_inputs=True)
-    # @TODO: decide where to store results
-    print(results)
+    with open(path, "rb") as fp:
+        err = cp.load(fp)
+        return err["error message"]
```
Comment (NicolasGensollen) [inline clinica/pydra/engine_utils.py:104]:
```suggestion
    return err["error message"]
```

File: clinica/pydra/engine_utils.py [hunk 4]
```diff
@@ -48,19 +52,53 @@ def bids_query(keys: list) -> dict:
...
+        with Submitter(plugin="cf") as submitter:
+            submitter(wf)
+    except Exception as e:
+        path = re.search("\/.*\.pklz", str(e))
+        if path:
+            path = path.group(0)
+            print(read_error(path))
+        else:
+            print("Exception:", e)
+        return str(e)
```
Comment (NicolasGensollen) [inline clinica/pydra/engine_utils.py:80]: I would raise the error instead of printing and returning it. This way we avoid the weird return type and make sure the execution stops. WDYT?

Comment (omar-rifai) [reply clinica/pydra/engine_utils.py:77]: I ended up doing a return directly to avoid the traceback messages. Seems to be the most readable version.

Comment (NicolasGensollen) [reply clinica/pydra/engine_utils.py:77]: @omar-rifai I was thinking about something along those lines (I haven't tested, there are probably some mistakes to fix and some changes to make...):

```python
class WorkflowExecutionError(Exception):
    def __init__(self, message="Error running the Workflow."):
        import re
        self.message = message
        path = re.search("\/.*\.pklz", self.message)
        if path:
            self.message = read_error(path.group(0))
        super().__init__(self.message)

def read_error(path: PathLike) -> str:
    with open(path, "rb") as fp:
        err = cp.load(fp)
    return err["error message"][:-1]

try:
    with Submitter(plugin="cf") as submitter:
        submitter(wf)
except Exception as e:
    raise WorkflowExecutionError(str(e))
```

Comment (omar-rifai) [reply clinica/pydra/engine_utils.py:77]: As discussed offline, we can leave this suggestion for a release version and keep the current version for debugging.

**Review by NicolasGensollen — APPROVED** [review]
Let's try it like this for debugging.
